File: leetcode/0605.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        cnt = 0
        curr = 0
        while curr <= len(flowerbed) - 1:
            prev = 0 if curr == 0 else flowerbed[curr - 1]
            if prev == 0 and flowerbed[curr] == 0:
                if curr == len(flowerbed) - 1:
                    cnt += 1
                    curr += 1
                elif flowerbed[curr + 1] == 0:
                    cnt += 1
                    curr += 2
                else:
                    curr += 3
                    
            else:
                curr += 1
            if cnt >= n:
                return True
        return False

File: leetcode/0605.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int cnt = 0, idx = 0;
        while(idx <= flowerbed.size() - 1){
            int prev = idx == 0? 0 : flowerbed[idx - 1];
            int next = idx == flowerbed.size() - 1 ? idx : idx + 1;
            if (prev == 0 && flowerbed[idx] == 0 && flowerbed[next] == 0)
            {
                cnt += 1;
                idx += 2;
            }
            else
            {
                idx += 1;
            }
            if (cnt >= n)
                return true;
        }
        return false;
    }
};
